__tests__/src/config.test.ts

```diff
@@ -74,19 +75,32 @@ describe("Given a configuration", () => {
 
     const config = configFactory.getConfiguration()
 
+    expect(config.shaArchive).toBeDefined()
     expect(config.shaArchive).toEqual("https://github.com/mpermar/vib-action-test/tarball/martinpe-patch-1")
   })
 
+  it("When a special character present in URL from 'tarball' onwards, GitHub Action encodes it", () => {
+    process.env.GITHUB_REF_NAME = "#artine-patch-1" // this is what rules
+    process.env.GITHUB_EVENT_PATH = path.join(root, "resources", "github-event-path-branch.json") // still will use env var above
+    
+    const config =  configFactory.getConfiguration()
+
+    expect(config.shaArchive).toBeDefined()
+    expect(config.shaArchive).toContain('https://github.com/mpermar/vib-action-test/tarball/%23artine-patch-1')
+
+  })
+
   it("When push from branch and both SHA archive and REF are set then sha is picked from SHA env", () => {
     process.env.GITHUB_SHA = "aacf48f14ed73e4b368ab66abf4742b0e9afae54" // this will be ignored
     process.env.GITHUB_REF_NAME = "martinpe-patch-1" // this is what rules
     process.env.GITHUB_EVENT_PATH = path.join(root, "resources", "github-event-path-branch.json") // still will use env var above
 
     const config = configFactory.getConfiguration()
 
+    expect(config.shaArchive).toBeDefined()
     expect(config.shaArchive).toEqual(
-      "https://github.com/mpermar/vib-action-test/tarball/aacf48f14ed73e4b368ab66abf4742b0e9afae54"
-    )
+        "https://github.com/mpermar/vib-action-test/tarball/aacf48f14ed73e4b368ab66abf4742b0e9afae54"
+      )  
```

Comment:
A minor change, remove additional blank spaces to keep it consistent with the rest of the code.